dev/kyuubi-extension-spark-3-1/src/main/scala/org/apache/kyuubi/sql/watchdog/ForcedMaxOutputRowsRule.scala

```diff
@@ -49,10 +56,19 @@ case class ForcedMaxOutputRowsRule(session: SparkSession) extends Rule[LogicalPl
 
     maxOutputRowsOpt match {
       case Some(forcedMaxOutputRows) => val supported = p match {
-          case _: Project => true
-          case _: Aggregate => true
-          case Limit(_, _) => true
-          case _ => false
+        case agg: Aggregate => agg match {
+          case Aggregate(_, Alias(_, "havingCondition")::Nil, _) => false
```

Comment:
I see what you are trying to fix, but it seems too hack.

Since this issue is from Spark and fix after 3.2.0, I prefer don't add Limit for `Aggregate` now. what do you think @i7xh  ?

Reply:
Hi, It's a long time for reply the issue for some reason, I update this PR  commit and updating the resolution, meanwhile，support `Union`, `Distinct` operator